sw/device/tests/kmac_entropy_test.c

```diff
@@ -0,0 +1,226 @@
+  CHECK_DIF_OK(dif_kmac_get_error(kmac_p, &err_reg));
+  return (err_reg == kDifErrorEntropyWaitTimerExpired);
+}
+
+bool test_main(void) {
+  LOG_INFO("Running KMAC ENTROPY test...");
+
+  dif_kmac_t kmac;
+  dif_kmac_operation_state_t kmac_operation_state;
+  bool is_kmac_locked;
+  bool expected_timeout_status;
+  bool observed_timeout_status;
+  uint32_t hash_ctr;
+  dif_kmac_error_t err_status;
+  kmac_edn_timeout_t timeout_test;
+  uint32_t out[DIGEST_LEN_KMAC_MAX];
+
+  // Initialize KMAC HWIP.
+  CHECK_DIF_OK(
+      dif_kmac_init(mmio_region_from_addr(TOP_EARLGREY_KMAC_BASE_ADDR), &kmac));
+
+  // Configure KMAC to use EDN entropy
+  dif_kmac_config_t config = (dif_kmac_config_t){
+      .entropy_mode = kDifKmacEntropyModeEdn,
+      .entropy_fast_process = false,
+      // In the last iteration, we expect to see that hash_ctr rewinds to 0
+      .entropy_hash_threshold = nr_of_tests,
+      .message_big_endian = false,
+      .output_big_endian = false,
+      .sideload = false,
+      .msg_mask = false,
+  };
+
+  // Handle string encoding
+  dif_kmac_customization_string_t str_buffer;
+  CHECK_DIF_OK(dif_kmac_customization_string_init(
+      kmac_test_vector.customization_string,
+      kmac_test_vector.customization_string_len, &str_buffer));
+
+  // When customization_string is empty, use NULL to activate empty str path
+  dif_kmac_customization_string_t *s_buff =
+      kmac_test_vector.customization_string_len == 0 ? NULL : &str_buffer;
+
+  size_t digest_l =
+      kmac_test_vector.digest_len_is_fixed ? kmac_test_vector.digest_len : 0;
+
+  for (size_t i = 0; i < nr_of_tests; i++) {
+    LOG_INFO("loop_ctr = %d", i);
+    timeout_test = test_timeout_vals[i];
+
+    config.entropy_prescaler = timeout_test.prescaler;
+    config.entropy_wait_timer = timeout_test.wait_timer;
+    expected_timeout_status = timeout_test.timeout_expected;
+
+    CHECK_DIF_OK(dif_kmac_configure(&kmac, config));
+
+    CHECK_DIF_OK(dif_kmac_mode_kmac_start(&kmac, &kmac_operation_state,
+                                          kmac_test_vector.mode, digest_l,
+                                          &kmac_test_vector.key, s_buff));
+
+    CHECK_DIF_OK(dif_kmac_absorb(&kmac, &kmac_operation_state,
+                                 kmac_test_vector.message,
+                                 kmac_test_vector.message_len, NULL));
+
+    CHECK(DIGEST_LEN_KMAC_MAX >= kmac_test_vector.digest_len);
+
+    // This is where timeout might happen, so we handle dif return manually
+    dif_result_t res = dif_kmac_squeeze(&kmac, &kmac_operation_state, out,
+                                        kmac_test_vector.digest_len, NULL);
+
+    // It is OK to get kDifError at this point because of possible timeout
+    CHECK(res == kDifOk || res == kDifError);
+    observed_timeout_status = is_edn_timeout(&kmac);
+
+    if (observed_timeout_status) {
+      LOG_INFO("EDN timed out.");
+    } else {
+      LOG_INFO("EDN seed received before timeout.");
```

Comment:
If there is no timeout error, can we add some checks (if not already):
to make sure there is no error or interrupts?
Caz I believe line 122 only checks if the err_code matches the timeout error code.
What if (very unlikely) there are some other errors?